### PR TITLE
Fix dupe k8s model detection

### PR DESCRIPTION
**Pull request by wallyworld**
## Description of change

Duplicate model detection for k8s models used to include the cloud name as well as the owner in the selection criteria. This was wrong as it allowed 2 models of the same name in state model.

There was no unit test for the functionality that was removed that used a different cloud name- there's an existing unit test for the standard dupe case.

## QA steps

Add a k8s model with the same name to different clouds and get an error.

## Bug reference

https://bugs.launchpad.net/juju/+bug/1768857

### CODE REVIEW

**Comment by wallyworld** [comment]
[It's possible to add two models with same name on different k8s](https://trello.com/c/BmHGYHmJ/775-its-possible-to-add-two-models-with-same-name-on-different-k8s)

File: state/model.go [hunk 1]
```diff
@@ -414,24 +414,16 @@ func (ctlr *Controller) NewModel(args ModelArgs) (_ *Model, _ *State, err error)
 		// the same "owner" and "name" in the collection. If the txn is
 		// aborted, check if it is due to the unique key restriction.
 		name := args.Config.Name()
-		qualifierTerm := bson.DocElem{"owner", owner.Id()}
-		if args.Type == ModelTypeCAAS {
```
Comment (ycliuhw) [inline state/model.go]: so for caas, we have owner as well, right?

**Comment by ycliuhw** [comment]
test failed

File: state/model.go (hunk 1 above)
Comment (wallyworld) [reply state/model.go]: yeah, all models have owners

**Comment by wallyworld** [comment]
$$merge$$

**Comment by wallyworld** [comment]
$$merge$$

**Pull request merged**